factoryWithTypeCheckers.js

```diff
@@ -7,14 +7,32 @@
 
 'use strict';
 
-var emptyFunction = require('fbjs/lib/emptyFunction');
-var invariant = require('fbjs/lib/invariant');
-var warning = require('fbjs/lib/warning');
 var assign = require('object-assign');
 
 var ReactPropTypesSecret = require('./lib/ReactPropTypesSecret');
 var checkPropTypes = require('./checkPropTypes');
 
+var printWarning = function() {};
+
+if (process.env.NODE_ENV !== 'production') {
+  printWarning = function(text) {
```

Comment:
Why copy this in each file? No big deal, since it's small.

Reply:
I thought about putting into `prop-types/warning` but then figured someone will start using *that* as a "public" API. Might as well just tuck it away.